ts/spi/provider_frame.ts

```diff
@@ -460,7 +460,7 @@ export class ProviderFrame {
       Promise<Credential[]> {
     // get all credentials across all domains; from this, we can filter down
     // to the set of credentials
-    let allCredentials = await this.credentialDataProvider.getAllCredentials();
+    let allCredentials = await this.credentialDataProvider.getAllHints(options);
```

Comment:
The design is potentially too opinionated; we could extract the code for de-duplication into a reusable module, and otherwise leave filtering in the hands of the data provider. WDYT?